dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,351 @@
-            String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
-            if(UtilMethods.isSet(indexName)) {
-                APILocator.getESIndexAPI().delete(indexName);
-            }
-            return Response.ok().build();
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to delete index: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
-        }
+        InitDataObject init=auth(httpServletRequest,httpServletResponse);
+        String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
+        return modIndex(httpServletRequest, httpServletResponse, indexName, IndexAction.CLEAR.name());
     }
 
-    @PUT
-    @Path("/activate/{params:.*}")
-    public Response activateIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
-        try {
-            InitDataObject init=auth(params,httpServletRequest, httpServletResponse);
-            String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
+    @CloseDBIfOpened
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/reindex")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getReindexationProgress(@Context final HttpServletRequest request,
+                    @Context final HttpServletResponse response) throws DotDataException {
 
-            activateIndex(indexName);
+        final InitDataObject init = auth(request, response);
+
+        return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
 
-            return Response.ok().build();
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to activate index: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
-        }
     }
+    private static final String DOTALL="DOTALL";
+    
+    
+    @CloseDBIfOpened
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/reindex")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response startReindex(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    @QueryParam("shards") int shards, @DefaultValue(DOTALL) @QueryParam("contentType") String contentType) throws DotDataException, DotSecurityException {
+        final InitDataObject init = auth(request, response);
+        shards = (shards <= 0) ? Config.getIntProperty("es.index.number_of_shards", 2) : shards;
+
+        System.setProperty("es.index.number_of_shards", String.valueOf(shards));
+        Logger.info(this, "Running Contentlet Reindex");
+
+        if(!DOTALL.equals(contentType)) {
+            ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(contentType);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'type' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)